src/components/TextInputBase/TextInputBase.style.ts

```diff
@@ -48,75 +67,31 @@ export const wrapperStyle = ({
   const error = status === 'error';
-  const backgroundColor = dark ? theme.utils.getColor('darkGrey', 750) : theme.palette.white;
 
-  return css`
-    transition: background-color 0.25s, box-shadow 0.25s, border-color 0.25s;
-    border-radius: ${isSearch ? rem(100) : theme.spacing.xsm};
-    cursor: ${disabled || locked ? 'not-allowed' : 'auto'};
-    flex: 1 1 100%;
-    user-select: none;
-    position: relative;
-    background-color: ${lean ? 'transparent' : backgroundColor};
-    opacity: ${disabled && 0.5};
-    border: ${rem(2)} solid transparent;
-    ${wrapperStyleSwitch(theme, lean, error, styleType, Boolean(disabled || locked))}
-    border-color: ${error ? theme.utils.getColor('error', 550, 'normal') : undefined};
-
-    /** This is added to prevent the field from growing/shrinking when the Clear icon shows/hides. 
-        The values used are the minimum widths of this field */ 
-    ${isSearch &&
-      Boolean(rightIcon) &&
-      `&{
-      min-width: ${size === 'sm' ? rem(286) : rem(264)}
-      }`}
-    
-    
-    ${!lean &&
-      !disabled &&
-      !locked &&
-      `&:hover {
-      background-color: ${theme.utils.getColor('lightGrey', 50)} !important;
-      border-color: ${styleType === 'outlined' && !error && theme.utils.getColor('lightGrey', 200)};
-      box-shadow: ${styleType === 'elevated' && theme.elevation['02']};
-    }
-    `}
-
-    &:focus-within {
-      border-color: ${!lean &&
-        !error &&
-        (isSearch ? theme.utils.getColor('blue', 550) : theme.utils.getColor('lightGrey', 200))};
-      box-shadow: ${styleType === 'elevated' && theme.elevation['02']};
-      background-color: ${
-        isSearch ? theme.utils.getColor('blue', null, 'pale') : theme.palette.white
-      };
-    }
-
-    ${(disabled || locked) &&
-      `
-      &:before {
-        content: '';
-        background-color: rgba(255, 255, 255, 0.15);
-        height: 100%;
-        width: 100%;
-        position: absolute;
-        left: 0;
-        top: 0;
-        z-index: 1;
-      }
-      > input, > textarea {
-        color: ${theme.utils.getColor('lightGrey', 750)};
-        fill: ${theme.utils.getColor('lightGrey', 750)};
-      }
-  `}
-  `;
+  return css({
+    transition: `background-color 0.25s, box-shadow 0.25s, border-color 0.25s`,
+    boxShadow: `0 0 0 ${rem(borderConfig.width)} ${
+      error
+        ? theme.utils.getColor(borderConfig.color.error.name, borderConfig.color.error.shade)
+        : theme.utils.getColor(borderConfig.color.default.name, borderConfig.color.default.shade)
+    }`,
+    borderRadius: isSearch ? rem(100) : theme.spacing.xsm,
+    flex: '1 1 100%',
+    userSelect: 'none',
+    position: 'relative',
+    opacity: disabled ? getDisabled().opacity : 1,
+    cursor: disabled || locked ? getDisabled().cursor : 'auto',
+    /** This is added to prevent the field from growing/shrinking when the Clear icon shows/hides.
+        The values used are the minimum widths of this field */
```

Comment:
Where can I see an example of this behaviour?

Reply:
This part was not refactored as it's completely out of the scope of the ticket.